internal/kokoro/check_incompat_changes.sh

```diff
@@ -21,33 +21,4 @@ if [[ `go version` != *"go1.16"* ]]; then
     exit 0
```

Comment:
@codyoss I could move this check into `shouldRun()` if you want to get rid of this script altogether. WDYT?